### PR TITLE
[Type Hints] `nn.GNNExplainer`

### PR DESCRIPTION
**Pull request by sebastian-montero**
@rusty1s Please note that I am using `torch.jit.export()` instead of `torch.jit.script()` for the tests. Can you please confirm if this is ok?

### CODE REVIEW

File: torch_geometric/nn/models/gnn_explainer.py [hunk 1]
```diff
@@ -97,7 +112,12 @@ def _clear_masks(self):
         self.node_feat_masks = None
         self.edge_mask = None
 
-    def _loss(self, log_logits, prediction, node_idx: Optional[int] = None):
+    def _loss(
+        self,
+        log_logits: Tensor,
+        prediction: Tensor,
+        node_idx: Optional[Tensor] = None,
+    ):
```
Comment (rusty1s) [inline torch_geometric/nn/models/gnn_explainer.py:120]:
```suggestion
    ) -> Tensor:
```

**Review by rusty1s — COMMENTED** [review]
`torch.jit.export` works here. Can we also add type hints to the function inside `GNNExplainer`?

**Comment by sebastian-montero** [comment]
> 

Hi @rusty1s, I think I already did. I added type hints to all methods in the class `GNNExplainer` here `torch_geometric/nn/models/gnn_explainer.py` 

Am I missing something?

**Comment by rusty1s** [comment]
Ah, sorry. Overlooked that single file, my bad.

**Comment by sebastian-montero** [comment]
> Ah, sorry. Overlooked that single file, my bad.

No problem! I'll resolve the Changelog conflicts and this should be ready to merge.

**Review by rusty1s — APPROVED** [review]

**Pull request merged**